pkg/util/validation/validate.go

```diff
@@ -55,6 +66,13 @@ var DiscardedSamples = prometheus.NewCounterVec(
 	},
 	[]string{discardReasonLabel, "user"},
 )
+var DiscardedMetadata = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "cortex_discarded_metadata_total",
+		Help: "The total number of metadata that were discarded.",
+	},
+	[]string{discardReasonLabel, "user"},
+)
 
 func init() {
 	prometheus.MustRegister(DiscardedSamples)
```

Comment:
You need to register `DiscardedMetadata` here as well.

Reply:
Thanks, good catch 🤦‍♂